### DISCUSSION TITLE
Custom number of outputs from spiking layer

### DISCUSSION THREAD

**Discussion by RFalowo** [fangwei123456/SpikingFlow#510] [Q&A]
I understand that the number of neurons in a given layer is equal to the input but i wanted to know if there is a way to set the output feature number?

**Comment by fangwei123456** [comment]
> if there is a way to set the output feature number?

I think you can add a layer to implement a shape transform on the outputs of neurons?

**Reply by RFalowo** [reply]
How?

**Reply by fangwei123456** [reply]
For example, when the input of the spiking neuron layer has a shape of `[T, N, C, H, W]`, then the output of the spiking neuron layer also has a shape of `[T, N, C, H, W]`. If you want to get a output with shape `[T, N, C, H // 2, W // 2]`, then you can add a pooling layer.

**Reply by RFalowo** [reply]
I was wondering more if you could do it in a way that still relies on learning a representation/tranformation like a linear layer in an ANN instead of just dimensionality reduction or feature summarization that pooling gives you.

**Reply by fangwei123456** [reply]
Sure, you can add a linear/conv layer after the spiking neuron layer. But the outputs will be float if you do not add one more spiking neuron layer after the linear/conv layer.

**Reply by RFalowo** [reply]
And by this you mean a torch layer, not a spikingjelly container? If so this wouldnt work for me as I need my network to be completely SNN based. Also many thanks for your help!

**Reply by fangwei123456** [reply]
> And by this you mean a torch layer, not a spikingjelly container?

You can also add SpikingJelly's modules after a spiking neuron layer.

**Reply by RFalowo** [reply]
Brilliant! and in regards to the using activation_based.layer.linear before neuron.LIFNode. Does this create a hybrid network and add normal ANN nodes to the network even in the case of using a container? for instance:

`net = layer.LinearRecurrentContainer(
    nn.Sequential(
        layer.Linear(in_features, out_features),
        neuron.LIFNode(),
    ),
    in_features, out_features)`

**Reply by RFalowo** [reply]
Apologies i think i just need an example of what you mean.

**Reply by fangwei123456** [reply]
> Does this create a hybrid network

Yes, but whether it is a problem depends on your purpose of using SNNs. If you seek for biological plausibility rather than deployment on neuromorphic chips, then the float values will not be a problem.
